beacon_node/rest_api/src/validator.rs

```diff
@@ -33,42 +32,44 @@ impl ValidatorDuty {
 
 /// HTTP Handler to retrieve a the duties for a set of validators during a particular epoch
 pub fn get_validator_duties<T: BeaconChainTypes + 'static>(req: Request<Body>) -> ApiResult {
-    // Get beacon state
-    let beacon_chain = req
-        .extensions()
-        .get::<Arc<BeaconChain<T>>>()
-        .ok_or_else(|| ApiError::ServerError("Beacon chain extension missing".to_string()))?;
-    let head_state = &beacon_chain.head().beacon_state;
+    let log = get_logger_from_request(&req);
+    slog::trace!(log, "Validator duties requested of API: {:?}", &req);
+    let beacon_chain = get_beacon_chain_from_request::<T>(&req)?;
+    let head_state = get_head_state(beacon_chain.clone())?;
```

Comment:
We _do_ need the state caches here. However, we don't need _all_ of them.

We only need the committee cache for a single epoch. See next comment.